File: frontend/src/types/loan_draw.ts
export interface LoanDrawPeriod {
  period_id: number;
  loan_draw_id: number;
  draw_number: number;
  draw_date?: string;
  amount: number;
}

export interface LoanDraw {
  loan_draw_id: number;
  project_id: number;
  building_id?: number | null;
  loan_number?: string;
  inspection_date?: string;
  draw_category: string;
  description: string;
  csi_code?: string;
  budget_amount: number;
  drawn_at_closing: number;
  draw_budget: number; // computed: budget_amount - drawn_at_closing
  sort_order: number;
  periods: LoanDrawPeriod[];
}

export interface LoanDrawCreate {
  project_id: number;
  building_id?: number | null;
  loan_number?: string;
  inspection_date?: string;
  draw_category: string;
  description: string;
  csi_code?: string;
  budget_amount: number;
  drawn_at_closing: number;
  sort_order?: number;
}

export interface LoanDrawSummary {
  category: DrawCategory;
  total_budget: number;
  total_drawn_at_closing: number;
  total_draw_budget: number;
  draws: LoanDraw[];
}

export type DrawCategory = 'land' | 'soft_costs' | 'bank_fees' | 'hard_costs';

export const DRAW_CATEGORY_LABELS: Record<DrawCategory, string> = {
  land: 'Land',
  soft_costs: 'Soft Costs',
  bank_fees: 'Bank Fees',
  hard_costs: 'Hard Costs',
};

File: backend/app/schemas/loan_draw.py
from pydantic import BaseModel, ConfigDict
from typing import Optional, List
from datetime import date


class LoanDrawPeriodCreate(BaseModel):
    draw_number: int
    draw_date: Optional[date] = None
    amount: float = 0.0


class LoanDrawPeriodUpdate(BaseModel):
    draw_number: Optional[int] = None
    draw_date: Optional[date] = None
    amount: Optional[float] = None


class LoanDrawPeriodResponse(BaseModel):
    period_id: int
    loan_draw_id: int
    draw_number: int
    draw_date: Optional[date] = None
    amount: float

    model_config = ConfigDict(from_attributes=True)


class LoanDrawCreate(BaseModel):
    project_id: int
    building_id: Optional[int] = None
    loan_number: Optional[str] = None
    inspection_date: Optional[date] = None
    draw_category: str
    description: str
    csi_code: Optional[str] = None
    budget_amount: float = 0.0
    drawn_at_closing: float = 0.0
    sort_order: int = 0


class LoanDrawUpdate(BaseModel):
    building_id: Optional[int] = None
    loan_number: Optional[str] = None
    inspection_date: Optional[date] = None
    draw_category: Optional[str] = None
    description: Optional[str] = None
    csi_code: Optional[str] = None
    budget_amount: Optional[float] = None
    drawn_at_closing: Optional[float] = None
    sort_order: Optional[int] = None


class LoanDrawResponse(BaseModel):
    loan_draw_id: int
    project_id: int
    building_id: Optional[int] = None
    loan_number: Optional[str] = None
    inspection_date: Optional[date] = None
    draw_category: str
    description: str
    csi_code: Optional[str] = None
    budget_amount: float
    drawn_at_closing: float
    sort_order: int
    draw_budget: Optional[float] = None

    model_config = ConfigDict(from_attributes=True)


class LoanDrawSchedule(BaseModel):
    loan_draw_id: int
    project_id: int
    building_id: Optional[int] = None
    loan_number: Optional[str] = None
    inspection_date: Optional[date] = None
    draw_category: str
    description: str
    csi_code: Optional[str] = None
    budget_amount: float
    drawn_at_closing: float
    sort_order: int
    periods: List[LoanDrawPeriodResponse] = []

    model_config = ConfigDict(from_attributes=True)
